### ISSUE TITLE
View::new_dyn panic

### ISSUE THREAD

**Issue by Ben-PH** [sycamore-rs/sycamore#453]
**Describe the bug**
We refactored some dynamic DOM construction to use `new_dyn`, to choose between two non-dyn DOM 

**To Reproduce**
Instead of:
```rust
#[component(PromoSycamoreBody<G>)]
pub(super) fn render_promo_body(promo_body: ReadSignal<PromoPageRouteSycamore>) -> View<G> {
	let class_sig = Signal::new(None);
	main()
		.dyn_attr("class", class_sig.handle())
		.dyn_child(move || match &*promo_body.get() {
			PromoPageRouteSycamore::About => {
				class_sig.set(Some("page about"));
				about::AboutBody::build(())
			}
			PromoPageRouteSycamore::Roadmap => {
				class_sig.set(Some("page roadmap"));
				roadmap::RoadmapBody::build(())
			}
		})
		.build()
}
```

write:
```rust
#[component(PromoSycamoreBody<G>)]
pub(super) fn render_promo_body(promo_body: ReadSignal<PromoPageRouteSycamore>) -> View<G> {
	View::new_dyn(move || match &*promo_body.get() {
		PromoPageRouteSycamore::About => {
			main().class("page about").child(about::AboutBody::build(())).build()
		}
		PromoPageRouteSycamore::Roadmap => {
			main().class("page roadmap").child(roadmap::RoadmapBody::build(())).build()
		}
	})
}

```

Console panic:

```
Uncaught (in promise) Error: `unwrap_throw` failed
    __wbindgen_throw https://localhost.tably.com:8443/roadmap:5841
    init https://localhost.tably.com:8443/roadmap:5872
    async* https://localhost.tably.com:8443/roadmap:6267
    <anonymous> https://localhost.tably.com:8443/roadmap:6280
```

**Environment**

Sycamore 0.7

**Comment by lukechu10** [comment]
Sorry, the two code snippets seem to be the same... Perhaps you forgot to update the second one?

**Comment by Ben-PH** [comment]
Updated. Well done, me :face_in_clouds:

**Comment by lukechu10** [comment]
If you don't mind, coud you please also share where the component is being used?

**Comment by lukechu10** [comment]
Sorry for taking so long. The issue is because in 0.7, `.class` cannot add multiple classes at the same time. In your case, you have two classes so you need something like `.class("page").class("about")`. It's a bit ugly but its fixed once you upgrade to 0.8.

**Comment by Ben-PH** [comment]
I see that now. Thanks for the heads up
